src/Microsoft.TestPlatform.PlatformAbstractions/uap10.0/System/PlatformThread.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.PlatformAbstractions
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.VisualStudio.TestPlatform.PlatformAbstractions.Interfaces;
+
+    public class PlatformThread : IThread
+    {
+        /// <inheritdoc/>
+        public void Run(Action action, PlatformApartmentState apartmentState, bool waitForCompletion)
+        {
+            if (apartmentState == PlatformApartmentState.STA)
```

Comment:
Should we throw exception here? Users generally tend to use same run settings for a session(IDE), although STA won't make sense here, we could simply ignore the apartment state, Log it, & continue running?